hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotManager.java

```diff
@@ -1214,11 +1358,27 @@ public void initialize(MasterServices master, MetricsMaster metricsMaster) throw
 
     this.coordinator = new ProcedureCoordinator(comms, tpool, timeoutMillis, wakeFrequency);
     this.executorService = master.getExecutorService();
+    this.verifyWorkerAssigner = new WorkerAssigner(master,
+      conf.getInt("hbase.snapshot.verify.task.max", 3),
+      new ProcedureEvent<>("snapshot-verify-worker-assigning"));
+    restoreUnfinishedSnapshotProcedure();
+    restoreWorkers();
     resetTempDir();
     snapshotHandlerChoreCleanerTask =
         scheduleThreadPool.scheduleAtFixedRate(this::cleanupSentinels, 10, 10, TimeUnit.SECONDS);
   }
 
+  private void restoreUnfinishedSnapshotProcedure() {
```

Comment:
I expected that we restore the map when replay proc wal, in the afterReplay method of the SnapshotProcedure.

So this is because we will create SnapshotManager after replaying the proc wals?

Will there be race if we scan all the procedures here? I think we need to make sure that the procedure executor has not been started yet?